cluster-scope/bundles/gpu-taint-management/taint.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+# GPU Taint Watcher Script
+
+# Initial tainting of GPU nodes
+kubectl get nodes -l node-role.kubernetes.io/worker -o json |
+jq -r '
+  .items[]
+  | select(.metadata.labels["nvidia.com/gpu.product"] != null)
+  | .metadata.name + " " + .metadata.labels["nvidia.com/gpu.product"]
+' | while read NODE PRODUCT; do
+  echo "Initial tainting: $NODE with GPU product $PRODUCT"
+  kubectl taint node "$NODE" "nvidia.com/gpu.product=$PRODUCT:NoSchedule" --overwrite=false 2>/dev/null
+done
+
+# Declare associative array to track last seen GPU product per node
+declare -A LAST_PRODUCT
+
+# Continuous monitoring loop
+while true; do
+  kubectl get nodes -l node-role.kubernetes.io/worker -o json |
+  jq -r '.items[].metadata.name' | while read NODE_NAME; do
```

Comment:
Using `-o name` gets you `kind/name` output, allow you to do this:

```suggestion
  kubectl get nodes -l node-role.kubernetes.io/worker -o name | cut -f2 -d/ |
  while read NODE_NAME; do
```